### PR TITLE
Azure AI Projects SDK

### PR DESCRIPTION
**Pull request by ShivangiReja**
This PR adds support for Azure AI Projects SDK.

### CODE REVIEW

**Comment by ShivangiReja** [comment]
@microsoft-github-policy-service rerun

**Review by KrzysztofCwalina — APPROVED** [review]
I just scanned it, as we need to release a beta soon, but we should do a through review of this codebase before we GA

File: eng/Packages.Data.props [hunk 1]
```diff
@@ -129,6 +129,7 @@
     <PackageReference Update="Azure.Storage.Queues" Version="12.19.1" />
     <PackageReference Update="Azure.Storage.Files.Shares" Version="12.19.1" />
     <PackageReference Update="Azure.AI.OpenAI" Version="2.0.0" />
+    <PackageReference Update="Azure.AI.Inference" Version="1.0.0-beta.2" />
```
Comment (jsquire) [inline eng/Packages.Data.props:132]: Can we bump these up so that the list in this section  is alphabetical by package name ?
```suggestion
    <PackageReference Update="Azure.AI.Inference" Version="1.0.0-beta.2" />    
    <PackageReference Update="Azure.AI.OpenAI" Version="2.0.0" />
```

File: sdk/ai/Azure.AI.Projects/README.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+var connectionString = Environment.GetEnvironmentVariable("AZURE_AI_CONNECTION_STRING");
+AIProjectClient projectClient = new AIProjectClient(connectionString, new DefaultAzureCredential());
+```
+
+Once `AIProjectClient` is created, you can call `GetXXXClient()` methods on this client to retrieve instances of specific sub-clients.
+
+## Examples
+
+### Agents
+Agents in the Azure AI Projects client library are designed to facilitate various interactions and operations within your AI projects. They serve as the core components that manage and execute tasks, leveraging different tools and resources to achieve specific goals. The following steps outline the typical sequence for interacting with agents:
```
Comment (jsquire) [inline sdk/ai/Azure.AI.Projects/README.md:52]:
```suggestion

Agents in the Azure AI Projects client library are designed to facilitate various interactions and operations within your AI projects. They serve as the core components that manage and execute tasks, leveraging different tools and resources to achieve specific goals. The following steps outline the typical sequence for interacting with agents:
```

File: sdk/ai/Azure.AI.Projects/README.md [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+
+Create an `AgentClient`
+```C# Snippet:OverviewCreateAgentClient
+var connectionString = Environment.GetEnvironmentVariable("AZURE_AI_CONNECTION_STRING");
```
Comment (jsquire) [inline sdk/ai/Azure.AI.Projects/README.md:56]: Was using a credential with a connection string intentional?   It's an odd pattern, so I wasn't sure if just an endpoint was meant for the environment variable.

File: sdk/ai/Azure.AI.Projects/src/Azure.AI.Projects.csproj [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Description>This is the Azure.AI.Client client library for developing .NET applications with rich experience.</Description>
```
Comment (jsquire) [inline sdk/ai/Azure.AI.Projects/src/Azure.AI.Projects.csproj:3]: Probably want to fill these out with custom values before doing the beta release.

**Review by jsquire — APPROVED** [review]

File: sdk/ai/Azure.AI.Projects/README.md (hunk 3 above)
Comment (ShivangiReja) [reply sdk/ai/Azure.AI.Projects/README.md:56]: I totally agree, it’s not the most conventional pattern but this is the pattern supported by the service, and it’s referred to as a connection string. It combines the `<Endpoint>`, `<Subscription>`, and `<ProjectName>` into one string.

File: eng/Packages.Data.props [hunk 5]
```diff
@@ -128,6 +128,7 @@
     <PackageReference Update="Azure.Storage.Blobs" Version="12.21.1" />
     <PackageReference Update="Azure.Storage.Queues" Version="12.19.1" />
     <PackageReference Update="Azure.Storage.Files.Shares" Version="12.19.1" />
+    <PackageReference Update="Azure.AI.Inference" Version="1.0.0-beta.2" />    
     <PackageReference Update="Azure.AI.OpenAI" Version="2.0.0" />
```
Comment (dargilco) [inline eng/Packages.Data.props:132]: This seems unrelated to the new SDK. Is this intentional?

File: sdk/ai/Azure.AI.Projects/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+# Release History
+
+## 1.0.0-beta.1 (Unreleased)
+
```
Comment (dargilco) [inline sdk/ai/Azure.AI.Projects/CHANGELOG.md:4]: Nit: I would already put a release date here.

File: eng/Packages.Data.props (hunk 5 above)
Comment (ShivangiReja) [reply eng/Packages.Data.props:132]: Yes, we have a common file where we add package references.

For Azure AI Projects, we take the dependency on [inference](https://github.com/Azure/azure-sdk-for-net/blob/feature/azure-ai-sdk/agents/sdk/ai/Azure.AI.Projects/src/Azure.AI.Projects.csproj#L18) for connection APIs.

File: sdk/ai/Azure.AI.Projects/CHANGELOG.md (hunk 6 above)
Comment (ShivangiReja) [reply sdk/ai/Azure.AI.Projects/CHANGELOG.md:4]: Not sure exactly when we are going to release the package. I'll update the date right before the release.

File: sdk/ai/Azure.AI.Projects/README.md [hunk 7]
```diff
@@ -0,0 +1,351 @@
...
+```
+
+## Key concepts
+
+### Create and authenticate the client
+
+To interact with Azure AI Projects, you’ll need to create an instance of `AIProjectClient`. Use the appropriate credential type from the Azure Identity library. For example, [DefaultAzureCredential][azure_identity_dac]:
+
+```C# Snippet:OverviewCreateClient
+var connectionString = Environment.GetEnvironmentVariable("AZURE_AI_CONNECTION_STRING");
```
Comment (dargilco) [inline sdk/ai/Azure.AI.Projects/README.md:37]: It would be good if we all use the same environment variables, so I can test a Python sample then a C# sample in the same console window. We use `PROJECT_CONNECTION_STRING`, since this is exactly the term shown in the AI Studio UI.

File: sdk/ai/Azure.AI.Projects/README.md [hunk 8]
```diff
@@ -0,0 +1,351 @@
...
+AIProjectClient projectClient = new AIProjectClient(connectionString, new DefaultAzureCredential());
+```
+
+Once `AIProjectClient` is created, you can call `GetXXXClient()` methods on this client to retrieve instances of specific sub-clients.
+
```
Comment (dargilco) [inline sdk/ai/Azure.AI.Projects/README.md:42]: ... you can call client methods of the form `GetXxxClient()`  to retrieve...

File: sdk/ai/Azure.AI.Projects/README.md [hunk 9]
```diff
@@ -0,0 +1,351 @@
...
+}
+while (runResponse.Value.Status == RunStatus.Queued
+    || runResponse.Value.Status == RunStatus.InProgress);
+```
+
+## Troubleshooting
+
+When you interact with Azure AI Projects using the .NET SDK, errors returned by the service correspond to the same HTTP status codes returned for REST API requests.
+
+For example, if you try to create a client using an endpoint that doesn't match your Azure AI Resource endpoint, a `404` error is returned, indicating `Resource Not Found`.
```
Comment (dargilco) [inline sdk/ai/Azure.AI.Projects/README.md:325]: Do service errors get converted to exceptions by the SDK? It's not clear from this text.

Comment (dargilco) [reply sdk/ai/Azure.AI.Projects/README.md:325]: I see it does, based on https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/ai/Azure.AI.Inference/README.md#exceptions . I would just copy the text from there and modify.

File: sdk/ai/Azure.AI.Projects/README.md [hunk 10]
```diff
@@ -0,0 +1,351 @@
...
+
```
Comment (dargilco) [inline sdk/ai/Azure.AI.Projects/README.md:326]: Do you know how to turn on SDK logs? That's a very useful tool for troubleshooting, and good to show here as part of troubleshooting section (or provide a link)

File: sdk/ai/Azure.AI.Projects/README.md [hunk 11]
```diff
@@ -0,0 +1,351 @@
+# Azure AI Projects client library for .NET
+Use the AI Projects client library to:
+
+* **Develop Agents using the Azure AI Agent Service**, leveraging an extensive ecosystem of models, tools, and capabilities from OpenAI, Microsoft, and other LLM providers. The Azure AI Agent Service enables the building of Agents for a wide range of generative AI use cases. The package is currently in preview.
+* **Enumerate connections** in your Azure AI Studio project and get connection properties.
+For example, get the inference endpoint URL and credentials associated with your Azure OpenAI connection.
+
```
Comment (dargilco) [inline sdk/ai/Azure.AI.Projects/README.md:7]: Consider adding a TOC, as this doc is (will be getting) long

File: sdk/ai/Azure.AI.Projects/tests/Samples/Inference/Sample_Embeddings.cs [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+
+public class Sample_Embeddings : SamplesBase<AIProjectsTestEnvironment>
+{
+    [Test]
+    public void BasicEmbedding()
+    {
+        var connectionString = TestEnvironment.AzureAICONNECTIONSTRING;
+        EmbeddingsClient embeddingsClient = new AIProjectClient(connectionString, new DefaultAzureCredential()).GetEmbeddingsClient();
+
+        var input = new List<string> { "King", "Queen", "Jack", "Page" };
```
Comment (dargilco) [inline sdk/ai/Azure.AI.Projects/tests/Samples/Inference/Sample_Embeddings.cs:22]: Please provide full sentences as example, instead of single words, otherwise it looks like only words are supported. See for example: https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ai/azure-ai-inference/samples/sample_embeddings.py

(I gave the same feedback to Travis at the time)

File: sdk/ai/Azure.AI.Projects/tests/Samples/Inference/Sample_Embeddings.cs [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+        var requestOptions = new EmbeddingsOptions(input);
+
+        Response<EmbeddingsResult> response = embeddingsClient.Embed(requestOptions);
```
Comment (dargilco) [inline sdk/ai/Azure.AI.Projects/tests/Samples/Inference/Sample_Embeddings.cs:25]: You need to specify the model name

**Review by dargilco — APPROVED** [review]

File: sdk/ai/Azure.AI.Projects/README.md (hunk 10 above)
Comment (ShivangiReja) [reply sdk/ai/Azure.AI.Projects/README.md:326]: Sure, I'll add https://learn.microsoft.com/en-us/dotnet/azure/sdk/logging link

File: sdk/ai/Azure.AI.Projects/tests/Samples/Inference/Sample_Embeddings.cs (hunk 13 above)
Comment (ShivangiReja) [reply sdk/ai/Azure.AI.Projects/tests/Samples/Inference/Sample_Embeddings.cs:30]: Updated!

I was just following the inference samples.
